specutils/io/asdf/extension.py

```diff
@@ -3,16 +3,25 @@
 """
 import os
 import urllib
+import warnings
 
 from asdf.util import filepath_to_url
 from asdf.extension import AsdfExtension
-
-from astropy.io.misc.asdf.extension import ASTROPY_SCHEMA_URI_BASE
+from asdf.exceptions import AsdfDeprecationWarning
 
 from .tags.spectra import *  # noqa
 from .types import _specutils_types
 
 
+with warnings.catch_warnings():
+    warnings.filterwarnings(
+        "ignore",
+        category=AsdfDeprecationWarning,
+        message=r".*from astropy.io.misc.asdf.* subclasses the deprecated CustomType .*",
+    )
+    from astropy.io.misc.asdf.extension import ASTROPY_SCHEMA_URI_BASE
```

Comment:
Can this just import from `asdf-astropy` while you are at it?

Reply:
No, that does not exist in asdf-astropy.

Reply:
Wait, what? But `astropy.io.misc.asdf` will go away. What should specutils use then?

Reply:
You don't need this for `asdf-astropy` to function. The entire extension system which supports `astropy.io.misc.asdf` and apparently `specutils` is deprecated and going away. If #1011 is resolved there is no need for this.

Reply:
To be clear, packages using `astropy.io.misc.asdf` cannot simply switch to using `asdf-astropy`. They will have to re-write their extensions to use the non-legacy ASDF extension system instead.